### ISSUE TITLE
cProfile is supported by twistd, but not documented as such

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3356]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#3356|
|Type|defect|
|Created|2008-07-14 10:16:36Z|

`twistd --help` output for the `--profiler` option is hard coded to include _hotshot_ and _profile_.  This obscures the fact that _cProfile_ will also work.  It'd probably be good if:

  * `twistd --help` mentioned _cProfile_
  * the help text for that option were generated automatically from the data structure that actually defines what's supported
  * case were not relevant (`twistd --profiler=cprofile` should work)

<details><summary>Searchable metadata</summary>

```
trac-id__3356 3356
type__defect defect
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1216030596000000 1216030596000000
changetime__1216888241000000 1216888241000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_1"></a>|@therve commented|
|-|-|

(In [24293]) Fix cProfile management.

Refs [#3355](https://github.com/twisted/twisted/issues/3355)
Refs [#3356](https://github.com/twisted/twisted/issues/3356)

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_2"></a>|@therve set status to `closed`|
|-|-|

(In [24330]) Merge default-profiler-3355

Author: therve
Reviewer: exarkun
Fixes [#3355](https://github.com/twisted/twisted/issues/3355)
Fixes [#3356](https://github.com/twisted/twisted/issues/3356)

Make hotshot explicitely the default profiler in AppProfiler, and add cProfile
to twistd --help.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
